packages/headless/src/features/folding/folding-state.ts

```diff
@@ -1,12 +1,27 @@
 import {Result} from '../../api/search/search/result';
 
+export const collectionMoreResultsAvailableBuffer = 5;
+
+export type CollectionId = string;
+
 export interface FoldedResult extends Result {
   /**
    * The children of this result sorted in the same order as the search results.
    */
   children: FoldedResult[];
 }
 
+export interface Collection extends FoldedResult {
+  /**
+   * Whether more results are available in the collection.
+   */
+  moreResultsAvailable: boolean;
+  /**
+   * Whether there is an ongoing query to add more results to the collection.
+   */
+  isLoadingMoreResults: boolean;
+}
```

Comment:
[Relevant discussion](https://github.com/coveo/ui-kit/pull/791/files#r629445412).

Should this be renamed `FoldedCollection`?

Reply:
IMO Yes